explorer-api/src/Explorer/Web/Api/Legacy/Util.hs

```diff
@@ -72,11 +76,33 @@ collapseTxGroup xs =
     [] -> error "collapseTxGroup: groupOn produced [] on non-empty list (impossible)"
     (x:_) -> (fst x, map snd xs)
 
-decodeTextAddress :: Text -> Either ExplorerError Address
+decodeTextAddress :: Text -> Either ExplorerError (Addr TPraosStandardCrypto)
```

Comment:
I recommend using `cardano-api`'s `Address` type and utilities for doing this. Additionally, `cardano-api` exposes a `SerialiseAddress` typeclass with functions, `serialiseAddress` and `deserialiseAddress`. 

However, this seems to work as well.

Reply:
That was my original intent, yet I wasn't quite sure of the burden of introduced the Cardano.Api.Typed as a dependency here (and all the cardano-api specifics types). Since the rest of the code is mostly using things from cardano-ledger, I thought the interop would be easier with cardano-ledger-specs directly. 

Yet I agree, in the short/medium term, I'd really like to revise this codebase to use cardano-api for most things and ditch the rest.